src/Service.Tests/CosmosTests/SamplerTests.cs

```diff
@@ -0,0 +1,213 @@
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+
+namespace Azure.DataApiBuilder.Service.Tests.CosmosTests
+{
+    /// <summary>
+    /// The <c>SamplerTests</c> class contains unit tests for the different sampling strategies used in the schema generation process.
+    /// These tests validate the behavior of the <c>TopNSampler</c>, <c>PartitionBasedSampler</c>, and <c>TimeBasedSampler</c> classes
+    /// to ensure they produce correct and expected results based on various configurations and parameters.
+    /// </summary>
+    /// <remarks>
+    /// The tests cover various scenarios and edge cases for each sampler type. They verify that the samplers correctly handle
+    /// different configurations, such as sampling count, partition keys, and time-based parameters. The tests also ensure that
+    /// the samplers interact correctly with the Cosmos DB containers and return results as expected.
+    /// </remarks>
+    [TestClass, TestCategory(TestCategory.COSMOSDBNOSQL)]
+    public class SamplerTests : TestBase
+    {
+        private Database _database;
+        private Container _containerWithNamePk;
+        private Container _containerWithIdPk;
+
+        private List<int> _sortedTimespansIdPk = new();
+        private List<int> _sortedTimespansNamePk = new();
+
+        private Mock<ILogger<SamplerTests>> _mockLogger = new();
+
+        private const string CONTAINER_NAME_ID_PK = "containerWithIdPk";
+        private const string CONTAINER_NAME_NAME_PK = "containerWithNamePk";
+
+        /// <summary>
+        /// Initializes the test environment by creating Cosmos DB containers and populating them with sample data.
+        /// This setup is required to ensure that the test cases have the necessary data and schema for accurate testing.
+        /// The method also includes wait times to ensure unique timestamp generation for each item, which is essential for sampling accuracy.
+        /// </summary>
+        [TestInitialize]
+        public async Task Initialize()
+        {
+            CosmosClientProvider cosmosClientProvider = _application.Services.GetService<CosmosClientProvider>();
+            CosmosClient cosmosClient = cosmosClientProvider.Clients[cosmosClientProvider.RuntimeConfigProvider.GetConfig().DefaultDataSourceName];
+            _database = await cosmosClient.CreateDatabaseIfNotExistsAsync(DATABASE_NAME);
+
+            _containerWithIdPk = await _database.CreateContainerIfNotExistsAsync(CONTAINER_NAME_ID_PK, "/id");
+            _containerWithNamePk = await _database.CreateContainerIfNotExistsAsync(CONTAINER_NAME_NAME_PK, "/name");
+
+            // Insert sample items with a delay to ensure unique _ts values for each item.
+            // This delay allows for accurate timestamp-based sampling in the test cases.
+            CreateItems(DATABASE_NAME, CONTAINER_NAME_ID_PK, 10, waitInMs: 1000);
```

Comment:
For test consistency, should you also clear out records from container when the container already exists? 
repro:
1. run test, executes setup, then tests
2. stop process, cleanup doesn't run.